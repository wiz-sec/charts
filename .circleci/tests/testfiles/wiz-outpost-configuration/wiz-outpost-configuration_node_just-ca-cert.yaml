agent:
  clientId: client-id
  clientSecret: client-secret
  enabled: true
httpProxyConfiguration:
  caCertificate: --- BEGIN CERTIFICATE --- cert.me --- END CERTIFICATE ---
  enabled: true
  name: wiz-http-proxy-configuration
  nodeConfigurationType: gcp-ubuntu
podAnnotations: null
