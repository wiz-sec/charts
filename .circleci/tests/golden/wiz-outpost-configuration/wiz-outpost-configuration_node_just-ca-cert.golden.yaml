---
# Source: wiz-outpost-configuration/templates/agent.secret.yaml
apiVersion: v1
kind: Secret
metadata:
  name: wiz-outpost-agent-creds
  labels:
    helm.sh/chart: "GOLDEN_STATIC_VALUE"
    app.kubernetes.io/name: wiz-outpost-configuration
    app.kubernetes.io/instance: release-test
    app.kubernetes.io/version: "0.0.1"
    app.kubernetes.io/managed-by: Helm

stringData:
  clientId: "client-id"
  clientSecret: "client-secret"
---
# Source: wiz-outpost-configuration/templates/httpProxyConfiguration.secret.yaml
apiVersion: v1
kind: Secret
type: Opaque
metadata:
  name: wiz-http-proxy-configuration
  labels:
    helm.sh/chart: "GOLDEN_STATIC_VALUE"
    app.kubernetes.io/name: wiz-outpost-configuration
    app.kubernetes.io/instance: release-test
    app.kubernetes.io/version: "0.0.1"
    app.kubernetes.io/managed-by: Helm
  annotations:
    meta.helm.sh/release-name: "release-test"
    meta.helm.sh/release-namespace: "release-helm-namespace"
data:
  caCertificate: "LS0tIEJFR0lOIENFUlRJRklDQVRFIC0tLQpjZXJ0Lm1lCi0tLSBFTkQgQ0VSVElGSUNBVEUgLS0tCg=="
---
# Source: wiz-outpost-configuration/templates/nodeHttpProxyConfiguration.gcp.ubuntu.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: wiz-http-proxy-configuration-entrypoint
  labels:
    helm.sh/chart: "GOLDEN_STATIC_VALUE"
    app.kubernetes.io/name: wiz-outpost-configuration
    app.kubernetes.io/instance: release-test
    app.kubernetes.io/version: "0.0.1"
    app.kubernetes.io/managed-by: Helm
  annotations:
    meta.helm.sh/release-name: "release-test"
    meta.helm.sh/release-namespace: "release-helm-namespace"
data:
  entrypoint.sh: |
    #!/usr/bin/env bash

    set -euo pipefail -x

    ROOT_MOUNT_DIR="${ROOT_MOUNT_DIR:-/root}"
    if chroot "${ROOT_MOUNT_DIR}" test -f /var/run/http-proxy-configured; then
      echo "Proxy already configured, skipping"
      exit 0
    fi
    chroot "${ROOT_MOUNT_DIR}" tee<<EOF /var/run/http-proxy-configured
    v1.0 $(date)
    EOF

    # Create the containerd and sandbox-image systemd directory
    chroot "${ROOT_MOUNT_DIR}" mkdir -p /lib/systemd/system/containerd.service.d

    # Set the proxy for future processes, and use as an include file
    chroot "${ROOT_MOUNT_DIR}" tee -a <<EOF /etc/environment
    EOF

    # Configure Containerd with the proxy
    chroot "${ROOT_MOUNT_DIR}" tee <<EOF /lib/systemd/system/containerd.service.d/http-proxy.conf >/dev/null
    [Service]    
    EnvironmentFile=/etc/environment
    EOF
    chroot "${ROOT_MOUNT_DIR}" mkdir -p /usr/local/share/ca-certificates/extra
    chroot "${ROOT_MOUNT_DIR}" tee <<EOF /usr/local/share/ca-certificates/extra/root.cert.crt
    --- BEGIN CERTIFICATE ---
    cert.me
    --- END CERTIFICATE ---
    EOF
    chroot "${ROOT_MOUNT_DIR}" update-ca-certificates

    # Log the proxy settings to confirm
    chroot "${ROOT_MOUNT_DIR}" tee <<EOF /var/log/startup-script.log
    Proxy settings applied
    EOF

    # Restart containerd to apply the new proxy settings
    chroot "${ROOT_MOUNT_DIR}" systemctl daemon-reload
    chroot "${ROOT_MOUNT_DIR}" systemctl restart containerd
---
# Source: wiz-outpost-configuration/templates/nodeHttpProxyConfiguration.gcp.ubuntu.yaml
apiVersion: apps/v1
kind: DaemonSet
metadata:
  name: node-initializer
  labels:
    app: default-init
spec:
  selector:
    matchLabels:
      app: default-init
  updateStrategy:
    type: RollingUpdate
  template:
    metadata:
      labels:
        name: node-initializer
        app: default-init
    spec:
      hostPID: true
      volumes:
        - name: root-mount
          hostPath:
            path: /
        - name: entrypoint
          configMap:
            name: wiz-http-proxy-configuration-entrypoint
            defaultMode: 0744
      initContainers:
        - image: "marketplace.gcr.io/google/ubuntu2204"
          name: node-initializer
          command: ["/scripts/entrypoint.sh"]
          env:
            - name: ROOT_MOUNT_DIR
              value: /root
          securityContext:
            privileged: true
          volumeMounts:
            - name: root-mount
              mountPath: /root
            - name: entrypoint
              mountPath: /scripts
      containers:
        - image: "gcr.io/google-containers/pause:2.0"
          name: pause
